src/vs/workbench/contrib/debug/browser/debugEditorContribution.ts

```diff
@@ -714,47 +714,91 @@ export class DebugEditorContribution implements IDebugEditorContribution {
 		};
 	}
 
-	private getWordToPositionsMap(): Map<string, Position[]> {
-		if (!this.wordToLineNumbersMap) {
-			this.wordToLineNumbersMap = new Map<string, Position[]>();
-			const model = this.editor.getModel();
-			if (!model) {
-				return this.wordToLineNumbersMap;
-			}
-
-			// For every word in every line, map its ranges for fast lookup
-			for (let lineNumber = 1, len = model.getLineCount(); lineNumber <= len; ++lineNumber) {
-				const lineContent = model.getLineContent(lineNumber);
-
-				// If line is too long then skip the line
-				if (lineContent.length > MAX_TOKENIZATION_LINE_LEN) {
-					continue;
-				}
-
-				model.forceTokenization(lineNumber);
+	private tokenizeViewPort(model: ITextModel): void {
+		this.editor.getVisibleRanges().forEach(visibleRange => {
```

Comment:
Can we extract a shared method for both `tokenizeViewPort` and `tokenizeCompleteDocument`? It can be like

```typescript
private tokenizeRange(model: ITextModel, start: Range, end: Range) {
	
}
```

Reply:
Done :)